### PR TITLE
OPDS generation command

### PR DESCRIPTION
**Pull request by samwilson**
Bug: T268424

### CODE REVIEW

**Review by Tpt — APPROVED** [review]

**Comment by dayllanmaza** [comment]
One thing I noticed is that the xhtml format fails but I guess that's beyond the scope of this patch. I had a couple of nits but I'm happy to merge this and address those later on so things go back to working like before

File: src/OpdsBuilder.php [hunk 1]
```diff
@@ -40,18 +40,23 @@ class OpdsBuilder {
 	 * @param string $lang
 	 * @param string $exportBasePath
 	 */
-	public function __construct( BookProvider $bookProvider, $lang, $exportBasePath = '' ) {
+	public function __construct( BookProvider $bookProvider, Api $api, string $lang, $exportBasePath = '' ) {
```
[the remark this answers is not in the document]
Comment (samwilson) [reply src/OpdsBuilder.php:42]: Done.

File: src/Command/OpdsCommand.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		$lang = $input->getOption( 'lang' );
+		$category = str_replace( ' ', '_', $input->getOption( 'category' ) );
+
+		date_default_timezone_set( 'UTC' );
+		$this->api->setLang( $lang );
+		$provider = new BookProvider( $this->api, [ 'categories' => false, 'images' => false ] );
+
+		$exportPath = 'https://wsexport.toolforge.org/';
```
[the remark this answers is not in the document]
Comment (samwilson) [reply src/Command/OpdsCommand.php]: Yep! Fixed.

**Comment by dayllanmaza** [comment]
I've created https://phabricator.wikimedia.org/T270441 to deal with the xhtml invalid format

**Pull request merged**